[keybinding] Broken tests on OS X

@simark Didn't you accidentally break the keybinding tests after cleaning up the stubbing in the code?

Travis Build:
https://travis-ci.org/theia-ide/theia/builds/387394117?utm_source=github_status&utm_medium=notification

Commit:
https://github.com/theia-ide/theia/pull/2014/commits/924647d31bc12fbdbecafbd8202cff55bd9ec92a

PR:
https://github.com/theia-ide/theia/pull/2014

Maybe I am just overlooking something and someone else maybe I broke the tests, but currently, they fail on both the CI and locally.

```
  keys api
    ✓ should parse a string to a KeyCode correctly
    1) should parse a string containing special modifiers to a KeyCode correctly
    ✓ should parse a string containing special modifiers to a KeyCode correctly (macOS)
    ✓ it should serialize a keycode properly with BACKQUOTE + M1
    ✓ it should serialize a keycode properly with a + M2 + M3
    ✓ the order of the modifiers should not matter when parsing the key code
    ✓ it should parse ctrl key properly on both OS X and other platforms
    ✓ it should serialize a keycode properly with a + M4
    ✓ it should parse a multi keycode keybinding
    ✓ it should parse a multi keycode keybinding with no modifiers
    ✓ it should compare keysequences properly
    ✓ it should be a modifier only
    ✓ it should be multiple modifiers only
  21 passing (94ms)
  1 failing
  1) keys api should parse a string containing special modifiers to a KeyCode correctly:
     AssertionError: expected [Function] to throw an error
      at Context.<anonymous> (packages/core/src/browser/keybinding.spec.ts:341:54)

```

Could you please verify it? I could test it locally on OS X. We cannot release a new version of Theia if the tests do not pass. Thanks!

@marcdumais-work, I am CCing you as well. AFAIK you were planning to release a new `3.11.0` version.

--- a/packages/core/src/browser/keybinding.spec.ts
+++ b/packages/core/src/browser/keybinding.spec.ts
@@ -31,7 +31,6 @@ const expect = chai.expect;
 let keybindingRegistry: KeybindingRegistry;
 let commandRegistry: CommandRegistry;
 let testContainer: Container;
-let stub: sinon.SinonStub;
 
 before(async () => {
     testContainer = new Container();
@@ -75,6 +74,8 @@ before(async () => {
 });
 
 describe('keybindings', () => {
+
+    let stub: sinon.SinonStub;
 
     before(() => {
         disableJSDOM = enableJSDOM();
@@ -330,6 +331,7 @@ describe("keys api", () => {
     });
 
     it("should parse a string containing special modifiers to a KeyCode correctly", () => {
+        const stub = sinon.stub(os, 'isOSX').value(false);
         const keycode = KeyCode.parse("ctrl+b");
         expect(keycode.ctrl).to.be.true;
         expect(keycode.key).is.equal(Key.KEY_B);
@@ -344,11 +346,12 @@ describe("keys api", () => {
         expect(keycodeCtrlOrCommand.meta).to.be.false;
         expect(keycodeCtrlOrCommand.ctrl).to.be.true;
         expect(keycodeCtrlOrCommand.key).is.equal(Key.KEY_B);
+        stub.restore();
     });
 
     it("should parse a string containing special modifiers to a KeyCode correctly (macOS)", () => {
         KeyCode.resetKeyBindings();
-        stub = sinon.stub(os, 'isOSX').value(true);
+        const stub = sinon.stub(os, 'isOSX').value(true);
         const keycode = KeyCode.parse("ctrl+b");
         expect(keycode.ctrl).to.be.true;
         expect(keycode.key).is.equal(Key.KEY_B);
@@ -370,19 +373,21 @@ describe("keys api", () => {
     });
 
     it("it should serialize a keycode properly with BACKQUOTE + M1", () => {
-        stub = sinon.stub(os, 'isOSX').value(true);
+        const stub = sinon.stub(os, 'isOSX').value(true);
         let keyCode = KeyCode.createKeyCode({ first: Key.BACKQUOTE, modifiers: [KeyModifier.CtrlCmd] });
         let keyCodeString = keyCode.toString();
         expect(keyCodeString).to.be.equal("meta+`");
         let parsedKeyCode = KeyCode.parse(keyCodeString);
         expect(KeyCode.equals(parsedKeyCode, keyCode)).to.be.true;
 
-        stub = sinon.stub(os, 'isOSX').value(false);
+        sinon.stub(os, 'isOSX').value(false);
         keyCode = KeyCode.createKeyCode({ first: Key.BACKQUOTE, modifiers: [KeyModifier.CtrlCmd] });
         keyCodeString = keyCode.toString();
         expect(keyCodeString).to.be.equal("ctrl+`");
         parsedKeyCode = KeyCode.parse(keyCodeString);
         expect(KeyCode.equals(parsedKeyCode, keyCode)).to.be.true;
+
+        stub.restore();
     });
 
     it("it should serialize a keycode properly with a + M2 + M3", () => {
@@ -419,19 +424,21 @@ describe("keys api", () => {
             code: Key.BACKQUOTE.code,
             ctrlKey: true,
         });
-        stub = sinon.stub(os, 'isOSX').value(true);
+        const stub = sinon.stub(os, 'isOSX').value(true);
         expect(KeyCode.createKeyCode(event).keystroke).to.be.equal('Backquote+M4');
-        stub = sinon.stub(os, 'isOSX').value(false);
+        sinon.stub(os, 'isOSX').value(false);
         expect(KeyCode.createKeyCode(event).keystroke).to.be.equal('Backquote+M1');
+        stub.restore();
     });
 
     it("it should serialize a keycode properly with a + M4", () => {
-        stub = sinon.stub(os, 'isOSX').value(true);
+        const stub = sinon.stub(os, 'isOSX').value(true);
         const keyCode = KeyCode.createKeyCode({ first: Key.KEY_A, modifiers: [KeyModifier.MacCtrl] });
         const keyCodeString = keyCode.toString();
         expect(keyCodeString).to.be.equal("ctrl+a");
         const parsedKeyCode = KeyCode.parse(keyCodeString);
         expect(KeyCode.equals(parsedKeyCode, keyCode)).to.be.true;
+        stub.restore();
     });
 
     it("it should parse a multi keycode keybinding", () => {

